zarrs/src/group.rs

```diff
@@ -344,17 +347,17 @@ impl<TStorage: ?Sized + ReadableStorageTraits + ListableStorageTraits> Group<TSt
     ///
     /// # Errors
     /// Returns [`ArrayCreateError`] if there is a storage error or any metadata is invalid.
-    pub fn child_arrays(&self, recursive: bool) -> Result<Vec<Array<TStorage>>, ArrayCreateError> {
-        self.children(recursive)?
+    pub fn child_arrays(&self) -> Result<Vec<Array<TStorage>>, ArrayCreateError> {
+        self.children(false)?
             .into_iter()
             .filter_map(|node| {
-                let name = node.name();
+                let path = node.path().to_string();
                 let metadata: NodeMetadata = node.into();
                 match metadata {
                     NodeMetadata::Array(metadata) => Some(Array::new_with_metadata(
                         self.storage.clone(),
-                        name.as_str(),
-                        metadata.clone(),
+                        path.as_str(),
+                        metadata,
```

Comment:
I assume it's preferable to clone the path rather than the metadata. (Could also make method that destructures a Node or make its members pub)